File: java-home-inventory-core/src/br/com/sailboat/homeinventory/core/interactor/product/GetProduct.java
package br.com.sailboat.homeinventory.core.interactor.product;

import br.com.sailboat.homeinventory.core.entity.Product;
import br.com.sailboat.homeinventory.core.interactor.UseCase;
import br.com.sailboat.homeinventory.core.repository.ProductRepository;

public class GetProduct implements UseCase<Product> {

    private ProductRepository productRepository;
    private Long productId;

    public GetProduct(ProductRepository productRepository, Long productId) {
        this.productRepository = productRepository;
        this.productId = productId;
    }

    @Override
    public Product execute() {
        return productRepository.findById(productId);
    }

}

File: android-home-inventory-app/app/src/main/kotlin/br/com/sailboat/homeinventory/domain/usecase/GetProduct.kt
package br.com.sailboat.homeinventory.domain.usecase

import br.com.sailboat.homeinventory.domain.entity.Product
import br.com.sailboat.homeinventory.domain.repository.ProductRepository
import javax.inject.Inject

class GetProduct @Inject constructor(private val productRepository: ProductRepository) : UseCase<Long, Product>() {

    override fun execute(productId: Long) = productRepository.getProduct(productId)

}